src/core/lib/iomgr/resolve_address.h

```diff
@@ -37,20 +37,33 @@ typedef struct {
   grpc_resolved_address* addrs;
 } grpc_resolved_addresses;
 
+typedef struct grpc_address_resolver_vtable {
```

Comment:
If we're re-working this anyway, how about converting it to a C++ API, rather than adding yet another C-style vtable that we're going to have to convert to C++ later?

Reply:
I agree, I've created #14693, and will follow up immediately following this PR.